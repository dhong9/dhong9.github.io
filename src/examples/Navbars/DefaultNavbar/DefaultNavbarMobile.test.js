import renderer from "react-test-renderer";
import DefaultNavbarMobile from "examples/Navbars/DefaultNavbar/DefaultNavbarMobile";

// Icons
import GitHubIcon from "@mui/icons-material/GitHub";

// Mocks
jest.mock("@mui/material/Container", () => {
  const { forwardRef } = jest.requireActual("react");
  return {
    __esModule: true,
    default: forwardRef(() => <div>MUI Container</div>),
  };
});
jest.mock("@mui/material/Icon", () => {
  const { forwardRef } = jest.requireActual("react");
  return {
    __esModule: true,
    default: forwardRef(() => <div>MUI Icon</div>),
  };
});
jest.mock("@mui/material/Popper", () => {
  const { forwardRef } = jest.requireActual("react");
  return {
    __esModule: true,
    default: forwardRef(() => <div>MUI Popper</div>),
  };
});
jest.mock("@mui/material/Grow", () => {
  const { forwardRef } = jest.requireActual("react");
  return {
    __esModule: true,
    default: forwardRef(() => <div>MUI Grow</div>),
  };
});
jest.mock("@mui/material/Grid", () => {
  const { forwardRef } = jest.requireActual("react");
  return {
    __esModule: true,
    default: forwardRef(() => <div>MUI Grid</div>),
  };
});
jest.mock("@mui/material/Divider", () => {
  const { forwardRef } = jest.requireActual("react");
  return {
    __esModule: true,
    default: forwardRef(() => <div>MUI Divider</div>),
  };
});
jest.mock("@mui/material/Link", () => {
  const { forwardRef } = jest.requireActual("react");
  return {
    __esModule: true,
    default: forwardRef(() => <div>MUI Link</div>),
  };
});
jest.mock("components/MKBox", () => {
  const { forwardRef } = jest.requireActual("react");
  const MKBox = forwardRef((props, ref) => (
    <div ref={ref}>
      Mock MKBox
      {props.children}
    </div>
  ));
  MKBox.propTypes = {}; // Add any necessary prop types here
  return {
    __esModule: true,
    default: MKBox,
  };
});
jest.mock("components/MKTypography", () => {
  const { forwardRef } = jest.requireActual("react");
  return {
    __esModule: true,
    default: forwardRef(() => <div>Mock Typography</div>),
  };
});
jest.mock("components/MKButton", () => {
  const { forwardRef } = jest.requireActual("react");
  return {
    __esModule: true,
    default: forwardRef(() => <button>Mock Button</button>),
  };
});

describe("DefaultNavbarMobile", () => {
  it("renders with route", () => {
    const component = renderer.create(
      <DefaultNavbarMobile
        routes={[{ label: "Navbar", route: "/default", icon: <GitHubIcon /> }]}
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders without route", () => {
    const component = renderer.create(
      <DefaultNavbarMobile routes={[{ label: "Regular", icon: <GitHubIcon /> }]} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
