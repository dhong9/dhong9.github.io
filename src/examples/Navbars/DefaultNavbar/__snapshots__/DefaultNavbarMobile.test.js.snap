// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultNavbarMobile handles collapse event 1`] = `
<div>
  <div
    class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
    style="min-height: 0px;"
  >
    <div
      class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
    >
      <div
        class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
      >
        <div
          class="MuiBox-root css-vib9bs"
        >
          <div
            class="MuiBox-root css-111f5ru"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-incl6n-MuiTypography-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-eimhud-MuiSvgIcon-root"
                data-testid="GitHubIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 1.27a11 11 0 00-3.48 21.46c.55.09.73-.28.73-.55v-1.84c-3.03.64-3.67-1.46-3.67-1.46-.55-1.29-1.28-1.65-1.28-1.65-.92-.65.1-.65.1-.65 1.1 0 1.73 1.1 1.73 1.1.92 1.65 2.57 1.2 3.21.92a2 2 0 01.64-1.47c-2.47-.27-5.04-1.19-5.04-5.5 0-1.1.46-2.1 1.2-2.84a3.76 3.76 0 010-2.93s.91-.28 3.11 1.1c1.8-.49 3.7-.49 5.5 0 2.1-1.38 3.02-1.1 3.02-1.1a3.76 3.76 0 010 2.93c.83.74 1.2 1.74 1.2 2.94 0 4.21-2.57 5.13-5.04 5.4.45.37.82.92.82 2.02v3.03c0 .27.1.64.73.55A11 11 0 0012 1.27"
                />
              </svg>
            </p>
            <span
              class="MuiTypography-root MuiTypography-button css-129xe1v-MuiTypography-root"
            >
              Github
            </span>
            <p
              class="MuiTypography-root MuiTypography-body2 css-wnhb9o-MuiTypography-root"
            >
              <span
                aria-hidden="true"
                class="material-icons-round notranslate MuiIcon-root MuiIcon-fontSizeInherit css-x5zaln-MuiIcon-root"
              />
            </p>
          </div>
          <a
            href="/rider"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-incl6n-MuiTypography-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-eimhud-MuiSvgIcon-root"
                data-testid="DonutSmallIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11 9.16V2c-5 .5-9 4.79-9 10s4 9.5 9 10v-7.16c-1-.41-2-1.52-2-2.84s1-2.43 2-2.84zM14.86 11H22c-.48-4.75-4-8.53-9-9v7.16c1 .3 1.52.98 1.86 1.84zM13 14.84V22c5-.47 8.52-4.25 9-9h-7.14c-.34.86-.86 1.54-1.86 1.84z"
                />
              </svg>
            </p>
            <span
              class="MuiTypography-root MuiTypography-button css-129xe1v-MuiTypography-root"
            >
              doughnut
            </span>
            <p
              class="MuiTypography-root MuiTypography-body2 css-wnhb9o-MuiTypography-root"
            >
              <span
                aria-hidden="true"
                class="material-icons-round notranslate MuiIcon-root MuiIcon-fontSizeInherit css-x5zaln-MuiIcon-root"
              >
                keyboard_arrow_down
              </span>
            </p>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`DefaultNavbarMobile renders with route 1`] = `
<div
  className="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
  style={
    {
      "minHeight": "0px",
    }
  }
>
  <div
    className="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
  >
    <div
      className="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
    >
      <div
        className="MuiBox-root css-vib9bs"
      >
        <a
          href="/default"
        >
          <p
            className="MuiTypography-root MuiTypography-body2 css-incl6n-MuiTypography-root"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-eimhud-MuiSvgIcon-root"
              data-testid="GitHubIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 1.27a11 11 0 00-3.48 21.46c.55.09.73-.28.73-.55v-1.84c-3.03.64-3.67-1.46-3.67-1.46-.55-1.29-1.28-1.65-1.28-1.65-.92-.65.1-.65.1-.65 1.1 0 1.73 1.1 1.73 1.1.92 1.65 2.57 1.2 3.21.92a2 2 0 01.64-1.47c-2.47-.27-5.04-1.19-5.04-5.5 0-1.1.46-2.1 1.2-2.84a3.76 3.76 0 010-2.93s.91-.28 3.11 1.1c1.8-.49 3.7-.49 5.5 0 2.1-1.38 3.02-1.1 3.02-1.1a3.76 3.76 0 010 2.93c.83.74 1.2 1.74 1.2 2.94 0 4.21-2.57 5.13-5.04 5.4.45.37.82.92.82 2.02v3.03c0 .27.1.64.73.55A11 11 0 0012 1.27"
              />
            </svg>
          </p>
          <span
            className="MuiTypography-root MuiTypography-button css-129xe1v-MuiTypography-root"
          >
            Github
          </span>
          <p
            className="MuiTypography-root MuiTypography-body2 css-wnhb9o-MuiTypography-root"
          >
            <span
              aria-hidden={true}
              className="material-icons-round notranslate MuiIcon-root MuiIcon-fontSizeInherit css-x5zaln-MuiIcon-root"
            />
          </p>
        </a>
        <a
          href="/rider"
        >
          <p
            className="MuiTypography-root MuiTypography-body2 css-incl6n-MuiTypography-root"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-eimhud-MuiSvgIcon-root"
              data-testid="DonutSmallIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M11 9.16V2c-5 .5-9 4.79-9 10s4 9.5 9 10v-7.16c-1-.41-2-1.52-2-2.84s1-2.43 2-2.84zM14.86 11H22c-.48-4.75-4-8.53-9-9v7.16c1 .3 1.52.98 1.86 1.84zM13 14.84V22c5-.47 8.52-4.25 9-9h-7.14c-.34.86-.86 1.54-1.86 1.84z"
              />
            </svg>
          </p>
          <span
            className="MuiTypography-root MuiTypography-button css-129xe1v-MuiTypography-root"
          >
            doughnut
          </span>
          <p
            className="MuiTypography-root MuiTypography-body2 css-wnhb9o-MuiTypography-root"
          >
            <span
              aria-hidden={true}
              className="material-icons-round notranslate MuiIcon-root MuiIcon-fontSizeInherit css-x5zaln-MuiIcon-root"
            >
              keyboard_arrow_down
            </span>
          </p>
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`DefaultNavbarMobile renders without route 1`] = `
<div
  className="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
  style={
    {
      "minHeight": "0px",
    }
  }
>
  <div
    className="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
  >
    <div
      className="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
    >
      <div
        className="MuiBox-root css-vib9bs"
      >
        <a
          href="/default"
        >
          <p
            className="MuiTypography-root MuiTypography-body2 css-incl6n-MuiTypography-root"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-eimhud-MuiSvgIcon-root"
              data-testid="GitHubIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 1.27a11 11 0 00-3.48 21.46c.55.09.73-.28.73-.55v-1.84c-3.03.64-3.67-1.46-3.67-1.46-.55-1.29-1.28-1.65-1.28-1.65-.92-.65.1-.65.1-.65 1.1 0 1.73 1.1 1.73 1.1.92 1.65 2.57 1.2 3.21.92a2 2 0 01.64-1.47c-2.47-.27-5.04-1.19-5.04-5.5 0-1.1.46-2.1 1.2-2.84a3.76 3.76 0 010-2.93s.91-.28 3.11 1.1c1.8-.49 3.7-.49 5.5 0 2.1-1.38 3.02-1.1 3.02-1.1a3.76 3.76 0 010 2.93c.83.74 1.2 1.74 1.2 2.94 0 4.21-2.57 5.13-5.04 5.4.45.37.82.92.82 2.02v3.03c0 .27.1.64.73.55A11 11 0 0012 1.27"
              />
            </svg>
          </p>
          <span
            className="MuiTypography-root MuiTypography-button css-129xe1v-MuiTypography-root"
          >
            Github
          </span>
          <p
            className="MuiTypography-root MuiTypography-body2 css-wnhb9o-MuiTypography-root"
          >
            <span
              aria-hidden={true}
              className="material-icons-round notranslate MuiIcon-root MuiIcon-fontSizeInherit css-x5zaln-MuiIcon-root"
            />
          </p>
        </a>
        <a
          href="/rider"
        >
          <p
            className="MuiTypography-root MuiTypography-body2 css-incl6n-MuiTypography-root"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-eimhud-MuiSvgIcon-root"
              data-testid="DonutSmallIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M11 9.16V2c-5 .5-9 4.79-9 10s4 9.5 9 10v-7.16c-1-.41-2-1.52-2-2.84s1-2.43 2-2.84zM14.86 11H22c-.48-4.75-4-8.53-9-9v7.16c1 .3 1.52.98 1.86 1.84zM13 14.84V22c5-.47 8.52-4.25 9-9h-7.14c-.34.86-.86 1.54-1.86 1.84z"
              />
            </svg>
          </p>
          <span
            className="MuiTypography-root MuiTypography-button css-129xe1v-MuiTypography-root"
          >
            doughnut
          </span>
          <p
            className="MuiTypography-root MuiTypography-body2 css-wnhb9o-MuiTypography-root"
          >
            <span
              aria-hidden={true}
              className="material-icons-round notranslate MuiIcon-root MuiIcon-fontSizeInherit css-x5zaln-MuiIcon-root"
            >
              keyboard_arrow_down
            </span>
          </p>
        </a>
      </div>
    </div>
  </div>
</div>
`;
